// @weareunite/unisys-angular-input-wrapper/esm5/lib/unisys-angular-input-wrapper.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { FormGroup } from "@angular/forms";
var UnisysAngularInputWrapperComponent = /** @class */ (function () {
    function UnisysAngularInputWrapperComponent() {
        this.customClass = '';
        this.fieldName = '';
    }
    /**
     * @return {?}
     */
    UnisysAngularInputWrapperComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.fieldName != '' ? this.field = this.group.controls[this.fieldName] : this.field = null;
    };
    /**
     * @param {?} val
     * @return {?}
     */
    UnisysAngularInputWrapperComponent.prototype.isString = /**
     * @param {?} val
     * @return {?}
     */
    function (val) {
        return typeof val === 'string';
    };
    UnisysAngularInputWrapperComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-unisys-angular-input-wrapper',
                    template: "<div class=\"form-group\" [formGroup]=\"group\" [ngClass]=\"customClass != '' && customClass\">\n    <label class=\"center-block\"> <span translate>{{ translation}}</span>\n        <ng-content></ng-content>\n    </label>\n    <ul *ngIf=\"fieldName\">\n        <li *ngFor=\"let item of field.errors | keyvalue\">\n            <span class=\"text-red\" style=\"font-size: 11px;font-weight: 700;\" translate>{{ item.key}}</span> <span *ngIf=\"isString(item.value)\" translate> {{ item.value}}</span>\n            <span *ngIf=\"!isString(item.value)\">\n          <ul>\n            <li *ngFor=\"let subitem of item.value | keyvalue\">\n              <span translate>{{ subitem.key}}</span> : {{ subitem.value}}\n            </li>\n          </ul>\n        </span>\n        </li>\n    </ul>\n</div>",
                    styles: ["label>span{font-weight:500;font-size:12px}.form-control.ng-invalid{border-color:#dd4b39}.form-control.ng-valid{border-color:#c3dccd}ul{padding-left:0;list-style:none}ul li ul{padding-left:5px}.error{font-size:11px;font-weight:700}"]
                },] },
    ];
    /** @nocollapse */
    UnisysAngularInputWrapperComponent.ctorParameters = function () { return []; };
    UnisysAngularInputWrapperComponent.propDecorators = {
        translation: [{ type: Input, args: ['translation',] }],
        group: [{ type: Input, args: ['group',] }],
        customClass: [{ type: Input, args: ['customClass',] }],
        fieldName: [{ type: Input, args: ['fieldName',] }]
    };
    return UnisysAngularInputWrapperComponent;
}());
export { UnisysAngularInputWrapperComponent };
if (false) {
    /** @type {?} */
    UnisysAngularInputWrapperComponent.prototype.translation;
    /** @type {?} */
    UnisysAngularInputWrapperComponent.prototype.group;
    /** @type {?} */
    UnisysAngularInputWrapperComponent.prototype.customClass;
    /** @type {?} */
    UnisysAngularInputWrapperComponent.prototype.fieldName;
    /** @type {?} */
    UnisysAngularInputWrapperComponent.prototype.field;
}
