// @weareunite/unisys-angular-input-wrapper/esm5/lib/unisys-angular-input-wrapper.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { CommonModule } from "@angular/common";
import { FormsModule, ReactiveFormsModule } from "@angular/forms";
import { NgModule } from '@angular/core';
import { UnisysAngularInputWrapperComponent } from './unisys-angular-input-wrapper.component';
import { HttpClient } from "@angular/common/http";
import { TranslateLoader, TranslateModule } from "@ngx-translate/core";
import { TranslateHttpLoader } from "@ngx-translate/http-loader";
/**
 * @param {?} http
 * @return {?}
 */
export function HttpLoaderFactory(http) {
    return new TranslateHttpLoader(http, './assets/i18n/', '.json');
}
var UnisysAngularInputWrapperModule = /** @class */ (function () {
    function UnisysAngularInputWrapperModule() {
    }
    UnisysAngularInputWrapperModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        TranslateModule.forRoot({
                            loader: {
                                provide: TranslateLoader,
                                useFactory: HttpLoaderFactory,
                                deps: [HttpClient]
                            }
                        }),
                    ],
                    declarations: [UnisysAngularInputWrapperComponent],
                    exports: [UnisysAngularInputWrapperComponent]
                },] },
    ];
    return UnisysAngularInputWrapperModule;
}());
export { UnisysAngularInputWrapperModule };
