// @weareunite/unisys-angular-input-wrapper/fesm5/weareunite-unisys-angular-input-wrapper.js
import { Injectable, Component, Input, defineInjectable, NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { HttpClient } from '@angular/common/http';
import { TranslateLoader, TranslateModule } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var UnisysAngularInputWrapperService = /** @class */ (function () {
    function UnisysAngularInputWrapperService() {
    }
    UnisysAngularInputWrapperService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    UnisysAngularInputWrapperService.ctorParameters = function () { return []; };
    /** @nocollapse */ UnisysAngularInputWrapperService.ngInjectableDef = defineInjectable({ factory: function UnisysAngularInputWrapperService_Factory() { return new UnisysAngularInputWrapperService(); }, token: UnisysAngularInputWrapperService, providedIn: "root" });
    return UnisysAngularInputWrapperService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var UnisysAngularInputWrapperComponent = /** @class */ (function () {
    function UnisysAngularInputWrapperComponent() {
        this.customClass = '';
        this.fieldName = '';
    }
    /**
     * @return {?}
     */
    UnisysAngularInputWrapperComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.fieldName != '' ? this.field = this.group.controls[this.fieldName] : this.field = null;
    };
    /**
     * @param {?} val
     * @return {?}
     */
    UnisysAngularInputWrapperComponent.prototype.isString = /**
     * @param {?} val
     * @return {?}
     */
    function (val) {
        return typeof val === 'string';
    };
    UnisysAngularInputWrapperComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-unisys-angular-input-wrapper',
                    template: "<div class=\"form-group\" [formGroup]=\"group\" [ngClass]=\"customClass != '' && customClass\">\n    <label class=\"center-block\"> <span translate>{{ translation}}</span>\n        <ng-content></ng-content>\n    </label>\n    <ul *ngIf=\"fieldName\">\n        <li *ngFor=\"let item of field.errors | keyvalue\">\n            <span class=\"text-red\" style=\"font-size: 11px;font-weight: 700;\" translate>{{ item.key}}</span> <span *ngIf=\"isString(item.value)\" translate> {{ item.value}}</span>\n            <span *ngIf=\"!isString(item.value)\">\n          <ul>\n            <li *ngFor=\"let subitem of item.value | keyvalue\">\n              <span translate>{{ subitem.key}}</span> : {{ subitem.value}}\n            </li>\n          </ul>\n        </span>\n        </li>\n    </ul>\n</div>",
                    styles: ["label>span{font-weight:500;font-size:12px}.form-control.ng-invalid{border-color:#dd4b39}.form-control.ng-valid{border-color:#c3dccd}ul{padding-left:0;list-style:none}ul li ul{padding-left:5px}.error{font-size:11px;font-weight:700}"]
                },] },
    ];
    /** @nocollapse */
    UnisysAngularInputWrapperComponent.ctorParameters = function () { return []; };
    UnisysAngularInputWrapperComponent.propDecorators = {
        translation: [{ type: Input, args: ['translation',] }],
        group: [{ type: Input, args: ['group',] }],
        customClass: [{ type: Input, args: ['customClass',] }],
        fieldName: [{ type: Input, args: ['fieldName',] }]
    };
    return UnisysAngularInputWrapperComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @param {?} http
 * @return {?}
 */
function HttpLoaderFactory(http) {
    return new TranslateHttpLoader(http, './assets/i18n/', '.json');
}
var UnisysAngularInputWrapperModule = /** @class */ (function () {
    function UnisysAngularInputWrapperModule() {
    }
    UnisysAngularInputWrapperModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        TranslateModule.forRoot({
                            loader: {
                                provide: TranslateLoader,
                                useFactory: HttpLoaderFactory,
                                deps: [HttpClient]
                            }
                        }),
                    ],
                    declarations: [UnisysAngularInputWrapperComponent],
                    exports: [UnisysAngularInputWrapperComponent]
                },] },
    ];
    return UnisysAngularInputWrapperModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { UnisysAngularInputWrapperService, UnisysAngularInputWrapperComponent, HttpLoaderFactory, UnisysAngularInputWrapperModule };
