// @weareunite/unisys-angular-input-wrapper/esm2015/lib/unisys-angular-input-wrapper.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { FormGroup } from "@angular/forms";
export class UnisysAngularInputWrapperComponent {
    constructor() {
        this.customClass = '';
        this.fieldName = '';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.fieldName != '' ? this.field = this.group.controls[this.fieldName] : this.field = null;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    isString(val) {
        return typeof val === 'string';
    }
}
UnisysAngularInputWrapperComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-unisys-angular-input-wrapper',
                template: `<div class="form-group" [formGroup]="group" [ngClass]="customClass != '' && customClass">
    <label class="center-block"> <span translate>{{ translation}}</span>
        <ng-content></ng-content>
    </label>
    <ul *ngIf="fieldName">
        <li *ngFor="let item of field.errors | keyvalue">
            <span class="text-red" style="font-size: 11px;font-weight: 700;" translate>{{ item.key}}</span> <span *ngIf="isString(item.value)" translate> {{ item.value}}</span>
            <span *ngIf="!isString(item.value)">
          <ul>
            <li *ngFor="let subitem of item.value | keyvalue">
              <span translate>{{ subitem.key}}</span> : {{ subitem.value}}
            </li>
          </ul>
        </span>
        </li>
    </ul>
</div>`,
                styles: [`label>span{font-weight:500;font-size:12px}.form-control.ng-invalid{border-color:#dd4b39}.form-control.ng-valid{border-color:#c3dccd}ul{padding-left:0;list-style:none}ul li ul{padding-left:5px}.error{font-size:11px;font-weight:700}`]
            },] },
];
/** @nocollapse */
UnisysAngularInputWrapperComponent.ctorParameters = () => [];
UnisysAngularInputWrapperComponent.propDecorators = {
    translation: [{ type: Input, args: ['translation',] }],
    group: [{ type: Input, args: ['group',] }],
    customClass: [{ type: Input, args: ['customClass',] }],
    fieldName: [{ type: Input, args: ['fieldName',] }]
};
if (false) {
    /** @type {?} */
    UnisysAngularInputWrapperComponent.prototype.translation;
    /** @type {?} */
    UnisysAngularInputWrapperComponent.prototype.group;
    /** @type {?} */
    UnisysAngularInputWrapperComponent.prototype.customClass;
    /** @type {?} */
    UnisysAngularInputWrapperComponent.prototype.fieldName;
    /** @type {?} */
    UnisysAngularInputWrapperComponent.prototype.field;
}
