// @weareunite/unisys-angular-form-group/esm5/lib/unisys-angular-form-group.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { FormGroup } from '@angular/forms';
var UnisysAngularFormGroupComponent = /** @class */ (function () {
    function UnisysAngularFormGroupComponent() {
        this.placeholderTranslation = '';
        this.type = 'input';
        this.rows = 1;
        this.customClass = '';
    }
    /**
     * @return {?}
     */
    UnisysAngularFormGroupComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.field = this.group.controls[this.fieldName];
    };
    /**
     * @param {?} val
     * @return {?}
     */
    UnisysAngularFormGroupComponent.prototype.isString = /**
     * @param {?} val
     * @return {?}
     */
    function (val) {
        return typeof val === 'string';
    };
    UnisysAngularFormGroupComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-unisys-angular-form-group',
                    template: "<div class=\"form-group\" [formGroup]=\"group\" [ngClass]=\"customClass != '' && customClass\">\n  <label class=\"center-block\">\n    <span translate>{{ translation}}</span>\n    <input *ngIf=\"type == 'input'\" class=\"form-control\" [formControlName]=\"fieldName\" [placeholder]=\"placeholderTranslation | translate\">\n    <textarea *ngIf=\"type == 'textarea'\" [rows]=\"rows\" class=\"form-control\" [formControlName]=\"fieldName\" [placeholder]=\"placeholderTranslation | translate\"></textarea>\n  </label>\n  <ul>\n    <li *ngFor=\"let item of field.errors | keyvalue\" class=\"error\">\n      <span class=\"text-red\" translate>{{ item.key}}</span>\n      <span *ngIf=\"isString(item.value)\" translate> {{ item.value}}</span>\n      <span *ngIf=\"!isString(item.value)\">\n          <ul>\n            <li *ngFor=\"let subitem of item.value | keyvalue\" class=\"error\">\n              <span translate>{{ subitem.key}}</span> : {{ subitem.value}}\n            </li>\n          </ul>\n        </span>\n    </li>\n  </ul>\n</div>\n\n\n",
                    styles: ["label>span{font-weight:500;font-size:12px}.form-control.ng-touched.ng-invalid{border-color:#dd4b39}.form-control.ng-touched.ng-valid{border-color:#c3dccd}ul{padding-left:0;list-style:none}ul li ul{padding-left:5px}.error{font-size:11px;font-weight:700}"]
                }] }
    ];
    /** @nocollapse */
    UnisysAngularFormGroupComponent.ctorParameters = function () { return []; };
    UnisysAngularFormGroupComponent.propDecorators = {
        translation: [{ type: Input, args: ['translation',] }],
        placeholderTranslation: [{ type: Input, args: ['placeholderTranslation',] }],
        fieldName: [{ type: Input, args: ['fieldName',] }],
        type: [{ type: Input, args: ['type',] }],
        rows: [{ type: Input, args: ['rows',] }],
        customClass: [{ type: Input, args: ['customClass',] }],
        group: [{ type: Input }]
    };
    return UnisysAngularFormGroupComponent;
}());
export { UnisysAngularFormGroupComponent };
if (false) {
    /** @type {?} */
    UnisysAngularFormGroupComponent.prototype.translation;
    /** @type {?} */
    UnisysAngularFormGroupComponent.prototype.placeholderTranslation;
    /** @type {?} */
    UnisysAngularFormGroupComponent.prototype.fieldName;
    /** @type {?} */
    UnisysAngularFormGroupComponent.prototype.type;
    /** @type {?} */
    UnisysAngularFormGroupComponent.prototype.rows;
    /** @type {?} */
    UnisysAngularFormGroupComponent.prototype.customClass;
    /** @type {?} */
    UnisysAngularFormGroupComponent.prototype.group;
    /** @type {?} */
    UnisysAngularFormGroupComponent.prototype.field;
}
