// @weareunite/unisys-angular-form-group/esm5/lib/unisys-angular-form-group.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { UnisysAngularFormGroupComponent } from './unisys-angular-form-group.component';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { TranslateLoader, TranslateModule } from '@ngx-translate/core';
import { HttpClient } from '@angular/common/http';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
/**
 * @param {?} http
 * @return {?}
 */
export function HttpLoaderFactory(http) {
    return new TranslateHttpLoader(http, './assets/i18n/', '.json');
}
var UnisysAngularFormGroupModule = /** @class */ (function () {
    function UnisysAngularFormGroupModule() {
    }
    UnisysAngularFormGroupModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [UnisysAngularFormGroupComponent],
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        TranslateModule.forRoot({
                            loader: {
                                provide: TranslateLoader,
                                useFactory: HttpLoaderFactory,
                                deps: [HttpClient]
                            }
                        }),
                    ],
                    exports: [UnisysAngularFormGroupComponent]
                },] }
    ];
    return UnisysAngularFormGroupModule;
}());
export { UnisysAngularFormGroupModule };
