// @weareunite/unisys-angular-input-wrapper/bundles/weareunite-unisys-angular-input-wrapper.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/forms'), require('@angular/common/http'), require('@ngx-translate/core'), require('@ngx-translate/http-loader')) :
    typeof define === 'function' && define.amd ? define('@weareunite/unisys-angular-input-wrapper', ['exports', '@angular/core', '@angular/common', '@angular/forms', '@angular/common/http', '@ngx-translate/core', '@ngx-translate/http-loader'], factory) :
    (factory((global.weareunite = global.weareunite || {}, global.weareunite['unisys-angular-input-wrapper'] = {}),global.ng.core,global.ng.common,global.ng.forms,global.ng.common.http,null,null));
}(this, (function (exports,i0,common,forms,http,core,httpLoader) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var UnisysAngularInputWrapperService = /** @class */ (function () {
        function UnisysAngularInputWrapperService() {
        }
        UnisysAngularInputWrapperService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] },
        ];
        /** @nocollapse */
        UnisysAngularInputWrapperService.ctorParameters = function () { return []; };
        /** @nocollapse */ UnisysAngularInputWrapperService.ngInjectableDef = i0.defineInjectable({ factory: function UnisysAngularInputWrapperService_Factory() { return new UnisysAngularInputWrapperService(); }, token: UnisysAngularInputWrapperService, providedIn: "root" });
        return UnisysAngularInputWrapperService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var UnisysAngularInputWrapperComponent = /** @class */ (function () {
        function UnisysAngularInputWrapperComponent() {
            this.customClass = '';
            this.fieldName = '';
        }
        /**
         * @return {?}
         */
        UnisysAngularInputWrapperComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.fieldName != '' ? this.field = this.group.controls[this.fieldName] : this.field = null;
            };
        /**
         * @param {?} val
         * @return {?}
         */
        UnisysAngularInputWrapperComponent.prototype.isString = /**
         * @param {?} val
         * @return {?}
         */
            function (val) {
                return typeof val === 'string';
            };
        UnisysAngularInputWrapperComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'lib-unisys-angular-input-wrapper',
                        template: "<div class=\"form-group\" [formGroup]=\"group\" [ngClass]=\"customClass != '' && customClass\">\n    <label class=\"center-block\"> <span translate>{{ translation}}</span>\n        <ng-content></ng-content>\n    </label>\n    <ul *ngIf=\"fieldName\">\n        <li *ngFor=\"let item of field.errors | keyvalue\">\n            <span class=\"text-red\" style=\"font-size: 11px;font-weight: 700;\" translate>{{ item.key}}</span> <span *ngIf=\"isString(item.value)\" translate> {{ item.value}}</span>\n            <span *ngIf=\"!isString(item.value)\">\n          <ul>\n            <li *ngFor=\"let subitem of item.value | keyvalue\">\n              <span translate>{{ subitem.key}}</span> : {{ subitem.value}}\n            </li>\n          </ul>\n        </span>\n        </li>\n    </ul>\n</div>",
                        styles: ["label>span{font-weight:500;font-size:12px}.form-control.ng-invalid{border-color:#dd4b39}.form-control.ng-valid{border-color:#c3dccd}ul{padding-left:0;list-style:none}ul li ul{padding-left:5px}.error{font-size:11px;font-weight:700}"]
                    },] },
        ];
        /** @nocollapse */
        UnisysAngularInputWrapperComponent.ctorParameters = function () { return []; };
        UnisysAngularInputWrapperComponent.propDecorators = {
            translation: [{ type: i0.Input, args: ['translation',] }],
            group: [{ type: i0.Input, args: ['group',] }],
            customClass: [{ type: i0.Input, args: ['customClass',] }],
            fieldName: [{ type: i0.Input, args: ['fieldName',] }]
        };
        return UnisysAngularInputWrapperComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /**
     * @param {?} http
     * @return {?}
     */
    function HttpLoaderFactory(http$$1) {
        return new httpLoader.TranslateHttpLoader(http$$1, './assets/i18n/', '.json');
    }
    var UnisysAngularInputWrapperModule = /** @class */ (function () {
        function UnisysAngularInputWrapperModule() {
        }
        UnisysAngularInputWrapperModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule,
                            forms.FormsModule,
                            forms.ReactiveFormsModule,
                            core.TranslateModule.forRoot({
                                loader: {
                                    provide: core.TranslateLoader,
                                    useFactory: HttpLoaderFactory,
                                    deps: [http.HttpClient]
                                }
                            }),
                        ],
                        declarations: [UnisysAngularInputWrapperComponent],
                        exports: [UnisysAngularInputWrapperComponent]
                    },] },
        ];
        return UnisysAngularInputWrapperModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    exports.UnisysAngularInputWrapperService = UnisysAngularInputWrapperService;
    exports.UnisysAngularInputWrapperComponent = UnisysAngularInputWrapperComponent;
    exports.HttpLoaderFactory = HttpLoaderFactory;
    exports.UnisysAngularInputWrapperModule = UnisysAngularInputWrapperModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
