// @weareunite/unisys-angular-form-group/bundles/weareunite-unisys-angular-form-group.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/forms'), require('@ngx-translate/core'), require('@angular/common/http'), require('@ngx-translate/http-loader')) :
    typeof define === 'function' && define.amd ? define('@weareunite/unisys-angular-form-group', ['exports', '@angular/core', '@angular/common', '@angular/forms', '@ngx-translate/core', '@angular/common/http', '@ngx-translate/http-loader'], factory) :
    (factory((global.weareunite = global.weareunite || {}, global.weareunite['unisys-angular-form-group'] = {}),global.ng.core,global.ng.common,global.ng.forms,global.core$1,global.ng.common.http,global.httpLoader));
}(this, (function (exports,core,common,forms,core$1,http,httpLoader) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var UnisysAngularFormGroupComponent = /** @class */ (function () {
        function UnisysAngularFormGroupComponent() {
            this.placeholderTranslation = '';
            this.type = 'input';
            this.rows = 1;
            this.customClass = '';
        }
        /**
         * @return {?}
         */
        UnisysAngularFormGroupComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.field = this.group.controls[this.fieldName];
            };
        /**
         * @param {?} val
         * @return {?}
         */
        UnisysAngularFormGroupComponent.prototype.isString = /**
         * @param {?} val
         * @return {?}
         */
            function (val) {
                return typeof val === 'string';
            };
        UnisysAngularFormGroupComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'lib-unisys-angular-form-group',
                        template: "<div class=\"form-group\" [formGroup]=\"group\" [ngClass]=\"customClass != '' && customClass\">\n  <label class=\"center-block\">\n    <span translate>{{ translation}}</span>\n    <input *ngIf=\"type == 'input'\" class=\"form-control\" [formControlName]=\"fieldName\" [placeholder]=\"placeholderTranslation | translate\">\n    <textarea *ngIf=\"type == 'textarea'\" [rows]=\"rows\" class=\"form-control\" [formControlName]=\"fieldName\" [placeholder]=\"placeholderTranslation | translate\"></textarea>\n  </label>\n  <ul>\n    <li *ngFor=\"let item of field.errors | keyvalue\" class=\"error\">\n      <span class=\"text-red\" translate>{{ item.key}}</span>\n      <span *ngIf=\"isString(item.value)\" translate> {{ item.value}}</span>\n      <span *ngIf=\"!isString(item.value)\">\n          <ul>\n            <li *ngFor=\"let subitem of item.value | keyvalue\" class=\"error\">\n              <span translate>{{ subitem.key}}</span> : {{ subitem.value}}\n            </li>\n          </ul>\n        </span>\n    </li>\n  </ul>\n</div>\n\n\n",
                        styles: ["label>span{font-weight:500;font-size:12px}.form-control.ng-touched.ng-invalid{border-color:#dd4b39}.form-control.ng-touched.ng-valid{border-color:#c3dccd}ul{padding-left:0;list-style:none}ul li ul{padding-left:5px}.error{font-size:11px;font-weight:700}"]
                    }] }
        ];
        /** @nocollapse */
        UnisysAngularFormGroupComponent.ctorParameters = function () { return []; };
        UnisysAngularFormGroupComponent.propDecorators = {
            translation: [{ type: core.Input, args: ['translation',] }],
            placeholderTranslation: [{ type: core.Input, args: ['placeholderTranslation',] }],
            fieldName: [{ type: core.Input, args: ['fieldName',] }],
            type: [{ type: core.Input, args: ['type',] }],
            rows: [{ type: core.Input, args: ['rows',] }],
            customClass: [{ type: core.Input, args: ['customClass',] }],
            group: [{ type: core.Input }]
        };
        return UnisysAngularFormGroupComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /**
     * @param {?} http
     * @return {?}
     */
    function HttpLoaderFactory(http$$1) {
        return new httpLoader.TranslateHttpLoader(http$$1, './assets/i18n/', '.json');
    }
    var UnisysAngularFormGroupModule = /** @class */ (function () {
        function UnisysAngularFormGroupModule() {
        }
        UnisysAngularFormGroupModule.decorators = [
            { type: core.NgModule, args: [{
                        declarations: [UnisysAngularFormGroupComponent],
                        imports: [
                            common.CommonModule,
                            forms.FormsModule,
                            forms.ReactiveFormsModule,
                            core$1.TranslateModule.forRoot({
                                loader: {
                                    provide: core$1.TranslateLoader,
                                    useFactory: HttpLoaderFactory,
                                    deps: [http.HttpClient]
                                }
                            }),
                        ],
                        exports: [UnisysAngularFormGroupComponent]
                    },] }
        ];
        return UnisysAngularFormGroupModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    exports.UnisysAngularFormGroupComponent = UnisysAngularFormGroupComponent;
    exports.HttpLoaderFactory = HttpLoaderFactory;
    exports.UnisysAngularFormGroupModule = UnisysAngularFormGroupModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
